hazelcast/src/main/java/com/hazelcast/nio/tcp/AbstractSelectionHandler.java

```diff
@@ -46,15 +47,43 @@ public AbstractSelectionHandler(TcpIpConnection connection, IOSelector ioSelecto
         this.initialOps = initialOps;
     }
 
+    public void migrate(final IOSelector newIOSelector) {
+        if (ioSelector == newIOSelector) {
+            return;
+        }
+
+        ioSelector.addTask(new Runnable() {
```

Comment:
Can you pull this into a regular inner class? I find anonymous inner classes always a PITA to read.